### PR TITLE
task 6 completed

### PR DESCRIPTION
**Pull request by Spoomgull**
added endpoint to retrieve all the comments from a specified article

### CODE REVIEW

File: __tests__/apiTests.test.js [hunk 1]
```diff
@@ -109,4 +109,59 @@ describe("GET /api/articles",()=>{
...
+    test("get status code 200, and return all the comments for the specified article",()=>{
+        return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+            .then(({body})=>{
+                const comments = body.comments
+                expect(comments.length).toBeGreaterThan(0)
+                comments.forEach((comment)=>{
+                    const commentParts = Object.values(comment)
+                    expect(Object.keys(comment)).toMatchObject(["comment_id","body","article_id","author","votes","created_at"])
```
Comment (roisinoreilly) [inline __tests__/apiTests.test.js:124]: this defeats the purpose of using .toMatchObject in my opinion to just assert against an array, and then use typeof for the next expect which is quite hard to read

File: __tests__/apiTests.test.js [hunk 2]
```diff
@@ -109,4 +109,59 @@ describe("GET /api/articles",()=>{
...
+    test("expect the returned values to be sorted with the most recent comments first",()=>{
+        return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+            .then(({body})=>{
+                const comments = body.comments
+                expect(comments).toBeSortedBy("created_at",{descending:true})
+            })
+        })
+    test("get status code 200, and return a message if there are no comments",()=>{
```
Comment (roisinoreilly) [inline __tests__/apiTests.test.js:143]: perfect

File: app/controller.js [hunk 3]
```diff
@@ -40,10 +40,27 @@ exports.getArticles = (req,res,next) =>{
...
+    const articleId = req.params.article_id
+
+    selectSpecifiedComments(articleId).then(({rows})=>{
+        const comments = rows
+ 
+        if(rows.length===0){return res.status(404).send({msg:"No comments!!"})}
+      
+        return res.status(200).send({comments:rows})
+
+    }).catch(({code})=>{if(code==="22P02"){
```
Comment (roisinoreilly) [inline app/controller.js:62]: this should be handled in your error handling middleware, you don't need to be concerned about error codes in your controller, you should just be passing them to the middleware to deal with it

**Review by roisinoreilly — COMMENTED** [review]
good start so far and great edge cases tested here, I've just left a few comments I'd like you to have a look at and go back and refactor. any questions let me know

File: __tests__/apiTests.test.js [hunk 4]
```diff
@@ -109,4 +109,59 @@ describe("GET /api/articles",()=>{
...
+        return request(app)
+        .get("/api/articles/2/comments")
+        .expect(404)
+            .then(({body})=>{
+                expect(body.msg).toBe("No comments!!")
+            })
+    })
+    test("get status code 404, and return an error message if the parameter is a valid type but a non-existant article_id",()=>{
```
Comment (roisinoreilly) [inline __tests__/apiTests.test.js:151]: this test seems good to me, but if you had a specific question about it give me a shout
